contrib/kind-dual-stack-conversion.sh

```diff
@@ -34,6 +34,7 @@ convert_cni() {
   # FIXME: kubectl rollout restart deployment leaves the old pod hanging 
```

Comment:
> To ensure my understanding is correct, running both modes together just means that we will have both running as same container in ovnkube-master pod right?

Correct.

Reply:
> Why is it recommended to run them separately? - the answer that comes to my mind is because in your following PRs, we are going to move the ovnkube-master pod to run as ovnkube-local-master container inside ovnkube-local-node pod on each node. Any other reasons?

That is the main reason.

For the deployments which don't want to enable IC,  it  does makes sense to run both the modes in the same container.    In my initial WIP PRs,  I had a config option in kind to run in both modes i.e as separate containers or both modes in one container.  I think I removed that support mainly due to failures in CI.  I found some issues related to watch factory or listeners.  I don't remember correctly.  But I can definitely revisit that.  Its not hard to add a job for that.

I agree that its better to add a job if we add "--init-cluster-manager" mode in ovnkube binary.

Otherwise we need to add a new binary for cluster-manager.  This could take a bit more work.

Let me know what's the preference.

Reply:
> running in same container is also a choice if users don't want to use IC?

Yes,  if the user passes both "--init-cluster-manager" and "--init-master".

But as I mentioned earlier,  unless we have a job to test this, we should not recommend.

I can remove this from the commit message if its confusing.

Reply:
To make things simple,  I'll update the code to only allow "--init-cluster-manager" or "--init-master" and not both.

Reply:
> That is the main reason.
> 
> For the deployments which don't want to enable IC, it does makes sense to run both the modes in the same container. In my initial WIP PRs, I had a config option in kind to run in both modes i.e as separate containers or both modes in one container. I think I removed that support mainly due to failures in CI. I found some issues related to watch factory or listeners. I don't remember correctly. But I can definitely revisit that. Its not hard to add a job for that.
> 
> I agree that its better to add a job if we add "--init-cluster-manager" mode in ovnkube binary.
> 
> Otherwise we need to add a new binary for cluster-manager. This could take a bit more work.
> 
> Let me know what's the preference.

To make things simple, I'll update the code to only allow "--init-cluster-manager" or "--init-master" but not both.
This would avoid confusion.  If there is a real value in supporting both the modes,  I think we can enhance it later.

Reply:
@trozet any thoughts ?

Reply:
microshift? They already run master and node in the same container and I would imagine they would like to have these two in the same container as well.

Reply:
Yeah, I was answering to @tssurya comment mainly.